// @aws-sdk/client-bedrock-agent/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@smithy/smithy-client";
import { BedrockAgentServiceException as __BaseException } from "./BedrockAgentServiceException";
export class AccessDeniedException extends __BaseException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
export const CustomControlMethod = {
    RETURN_CONTROL: "RETURN_CONTROL",
};
export var ActionGroupExecutor;
(function (ActionGroupExecutor) {
    ActionGroupExecutor.visit = (value, visitor) => {
        if (value.lambda !== undefined)
            return visitor.lambda(value.lambda);
        if (value.customControl !== undefined)
            return visitor.customControl(value.customControl);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(ActionGroupExecutor || (ActionGroupExecutor = {}));
export class ConflictException extends __BaseException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
export const ActionGroupState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export var APISchema;
(function (APISchema) {
    APISchema.visit = (value, visitor) => {
        if (value.s3 !== undefined)
            return visitor.s3(value.s3);
        if (value.payload !== undefined)
            return visitor.payload(value.payload);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(APISchema || (APISchema = {}));
export const Type = {
    ARRAY: "array",
    BOOLEAN: "boolean",
    INTEGER: "integer",
    NUMBER: "number",
    STRING: "string",
};
export const RequireConfirmation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export var FunctionSchema;
(function (FunctionSchema) {
    FunctionSchema.visit = (value, visitor) => {
        if (value.functions !== undefined)
            return visitor.functions(value.functions);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(FunctionSchema || (FunctionSchema = {}));
export const ActionGroupSignature = {
    AMAZON_CODEINTERPRETER: "AMAZON.CodeInterpreter",
    AMAZON_USERINPUT: "AMAZON.UserInput",
    ANTHROPIC_BASH: "ANTHROPIC.Bash",
    ANTHROPIC_COMPUTER: "ANTHROPIC.Computer",
    ANTHROPIC_TEXTEDITOR: "ANTHROPIC.TextEditor",
};
export class InternalServerException extends __BaseException {
    name = "InternalServerException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
}
export class ResourceNotFoundException extends __BaseException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
export class ServiceQuotaExceededException extends __BaseException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
export class ThrottlingException extends __BaseException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
export class ValidationException extends __BaseException {
    name = "ValidationException";
    $fault = "client";
    fieldList;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.fieldList = opts.fieldList;
    }
}
export const AgentCollaboration = {
    DISABLED: "DISABLED",
    SUPERVISOR: "SUPERVISOR",
    SUPERVISOR_ROUTER: "SUPERVISOR_ROUTER",
};
export const AgentStatus = {
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    NOT_PREPARED: "NOT_PREPARED",
    PREPARED: "PREPARED",
    PREPARING: "PREPARING",
    UPDATING: "UPDATING",
    VERSIONING: "VERSIONING",
};
export var OrchestrationExecutor;
(function (OrchestrationExecutor) {
    OrchestrationExecutor.visit = (value, visitor) => {
        if (value.lambda !== undefined)
            return visitor.lambda(value.lambda);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(OrchestrationExecutor || (OrchestrationExecutor = {}));
export const MemoryType = {
    SESSION_SUMMARY: "SESSION_SUMMARY",
};
export const OrchestrationType = {
    CUSTOM_ORCHESTRATION: "CUSTOM_ORCHESTRATION",
    DEFAULT: "DEFAULT",
};
export const CreationMode = {
    DEFAULT: "DEFAULT",
    OVERRIDDEN: "OVERRIDDEN",
};
export const PromptState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const PromptType = {
    KNOWLEDGE_BASE_RESPONSE_GENERATION: "KNOWLEDGE_BASE_RESPONSE_GENERATION",
    MEMORY_SUMMARIZATION: "MEMORY_SUMMARIZATION",
    ORCHESTRATION: "ORCHESTRATION",
    POST_PROCESSING: "POST_PROCESSING",
    PRE_PROCESSING: "PRE_PROCESSING",
};
export const AgentAliasStatus = {
    CREATING: "CREATING",
    DELETING: "DELETING",
    DISSOCIATED: "DISSOCIATED",
    FAILED: "FAILED",
    PREPARED: "PREPARED",
    UPDATING: "UPDATING",
};
export const RelayConversationHistory = {
    DISABLED: "DISABLED",
    TO_COLLABORATOR: "TO_COLLABORATOR",
};
export const KnowledgeBaseState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const DataDeletionPolicy = {
    DELETE: "DELETE",
    RETAIN: "RETAIN",
};
export const CrawlFilterConfigurationType = {
    PATTERN: "PATTERN",
};
export const ConfluenceAuthType = {
    BASIC: "BASIC",
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
};
export const ConfluenceHostType = {
    SAAS: "SAAS",
};
export const SalesforceAuthType = {
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
};
export const SharePointAuthType = {
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
    OAUTH2_SHAREPOINT_APP_ONLY_CLIENT_CREDENTIALS: "OAUTH2_SHAREPOINT_APP_ONLY_CLIENT_CREDENTIALS",
};
export const SharePointHostType = {
    ONLINE: "ONLINE",
};
export const DataSourceType = {
    CONFLUENCE: "CONFLUENCE",
    CUSTOM: "CUSTOM",
    REDSHIFT_METADATA: "REDSHIFT_METADATA",
    S3: "S3",
    SALESFORCE: "SALESFORCE",
    SHAREPOINT: "SHAREPOINT",
    WEB: "WEB",
};
export const WebScopeType = {
    HOST_ONLY: "HOST_ONLY",
    SUBDOMAINS: "SUBDOMAINS",
};
export const ChunkingStrategy = {
    FIXED_SIZE: "FIXED_SIZE",
    HIERARCHICAL: "HIERARCHICAL",
    NONE: "NONE",
    SEMANTIC: "SEMANTIC",
};
export const EnrichmentStrategyMethod = {
    CHUNK_ENTITY_EXTRACTION: "CHUNK_ENTITY_EXTRACTION",
};
export const ContextEnrichmentType = {
    BEDROCK_FOUNDATION_MODEL: "BEDROCK_FOUNDATION_MODEL",
};
export const StepType = {
    POST_CHUNKING: "POST_CHUNKING",
};
export const ParsingModality = {
    MULTIMODAL: "MULTIMODAL",
};
export const ParsingStrategy = {
    BEDROCK_DATA_AUTOMATION: "BEDROCK_DATA_AUTOMATION",
    BEDROCK_FOUNDATION_MODEL: "BEDROCK_FOUNDATION_MODEL",
};
export const DataSourceStatus = {
    AVAILABLE: "AVAILABLE",
    DELETE_UNSUCCESSFUL: "DELETE_UNSUCCESSFUL",
    DELETING: "DELETING",
};
export var FlowConnectionConfiguration;
(function (FlowConnectionConfiguration) {
    FlowConnectionConfiguration.visit = (value, visitor) => {
        if (value.data !== undefined)
            return visitor.data(value.data);
        if (value.conditional !== undefined)
            return visitor.conditional(value.conditional);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(FlowConnectionConfiguration || (FlowConnectionConfiguration = {}));
export const FlowConnectionType = {
    CONDITIONAL: "Conditional",
    DATA: "Data",
};
export var PromptInferenceConfiguration;
(function (PromptInferenceConfiguration) {
    PromptInferenceConfiguration.visit = (value, visitor) => {
        if (value.text !== undefined)
            return visitor.text(value.text);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(PromptInferenceConfiguration || (PromptInferenceConfiguration = {}));
export const CachePointType = {
    DEFAULT: "default",
};
export var ContentBlock;
(function (ContentBlock) {
    ContentBlock.visit = (value, visitor) => {
        if (value.text !== undefined)
            return visitor.text(value.text);
        if (value.cachePoint !== undefined)
            return visitor.cachePoint(value.cachePoint);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(ContentBlock || (ContentBlock = {}));
export const ConversationRole = {
    ASSISTANT: "assistant",
    USER: "user",
};
export var SystemContentBlock;
(function (SystemContentBlock) {
    SystemContentBlock.visit = (value, visitor) => {
        if (value.text !== undefined)
            return visitor.text(value.text);
        if (value.cachePoint !== undefined)
            return visitor.cachePoint(value.cachePoint);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(SystemContentBlock || (SystemContentBlock = {}));
export var ToolChoice;
(function (ToolChoice) {
    ToolChoice.visit = (value, visitor) => {
        if (value.auto !== undefined)
            return visitor.auto(value.auto);
        if (value.any !== undefined)
            return visitor.any(value.any);
        if (value.tool !== undefined)
            return visitor.tool(value.tool);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(ToolChoice || (ToolChoice = {}));
export var ToolInputSchema;
(function (ToolInputSchema) {
    ToolInputSchema.visit = (value, visitor) => {
        if (value.json !== undefined)
            return visitor.json(value.json);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(ToolInputSchema || (ToolInputSchema = {}));
export var Tool;
(function (Tool) {
    Tool.visit = (value, visitor) => {
        if (value.toolSpec !== undefined)
            return visitor.toolSpec(value.toolSpec);
        if (value.cachePoint !== undefined)
            return visitor.cachePoint(value.cachePoint);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(Tool || (Tool = {}));
export var PromptTemplateConfiguration;
(function (PromptTemplateConfiguration) {
    PromptTemplateConfiguration.visit = (value, visitor) => {
        if (value.text !== undefined)
            return visitor.text(value.text);
        if (value.chat !== undefined)
            return visitor.chat(value.chat);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(PromptTemplateConfiguration || (PromptTemplateConfiguration = {}));
export const PromptTemplateType = {
    CHAT: "CHAT",
    TEXT: "TEXT",
};
export var PromptFlowNodeSourceConfiguration;
(function (PromptFlowNodeSourceConfiguration) {
    PromptFlowNodeSourceConfiguration.visit = (value, visitor) => {
        if (value.resource !== undefined)
            return visitor.resource(value.resource);
        if (value.inline !== undefined)
            return visitor.inline(value.inline);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(PromptFlowNodeSourceConfiguration || (PromptFlowNodeSourceConfiguration = {}));
export var RetrievalFlowNodeServiceConfiguration;
(function (RetrievalFlowNodeServiceConfiguration) {
    RetrievalFlowNodeServiceConfiguration.visit = (value, visitor) => {
        if (value.s3 !== undefined)
            return visitor.s3(value.s3);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(RetrievalFlowNodeServiceConfiguration || (RetrievalFlowNodeServiceConfiguration = {}));
export var StorageFlowNodeServiceConfiguration;
(function (StorageFlowNodeServiceConfiguration) {
    StorageFlowNodeServiceConfiguration.visit = (value, visitor) => {
        if (value.s3 !== undefined)
            return visitor.s3(value.s3);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(StorageFlowNodeServiceConfiguration || (StorageFlowNodeServiceConfiguration = {}));
export var FlowNodeConfiguration;
(function (FlowNodeConfiguration) {
    FlowNodeConfiguration.visit = (value, visitor) => {
        if (value.input !== undefined)
            return visitor.input(value.input);
        if (value.output !== undefined)
            return visitor.output(value.output);
        if (value.knowledgeBase !== undefined)
            return visitor.knowledgeBase(value.knowledgeBase);
        if (value.condition !== undefined)
            return visitor.condition(value.condition);
        if (value.lex !== undefined)
            return visitor.lex(value.lex);
        if (value.prompt !== undefined)
            return visitor.prompt(value.prompt);
        if (value.lambdaFunction !== undefined)
            return visitor.lambdaFunction(value.lambdaFunction);
        if (value.storage !== undefined)
            return visitor.storage(value.storage);
        if (value.agent !== undefined)
            return visitor.agent(value.agent);
        if (value.retrieval !== undefined)
            return visitor.retrieval(value.retrieval);
        if (value.iterator !== undefined)
            return visitor.iterator(value.iterator);
        if (value.collector !== undefined)
            return visitor.collector(value.collector);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(FlowNodeConfiguration || (FlowNodeConfiguration = {}));
export const FlowNodeIODataType = {
    ARRAY: "Array",
    BOOLEAN: "Boolean",
    NUMBER: "Number",
    OBJECT: "Object",
    STRING: "String",
};
export const FlowNodeType = {
    AGENT: "Agent",
    COLLECTOR: "Collector",
    CONDITION: "Condition",
    INPUT: "Input",
    ITERATOR: "Iterator",
    KNOWLEDGE_BASE: "KnowledgeBase",
    LAMBDA_FUNCTION: "LambdaFunction",
    LEX: "Lex",
    OUTPUT: "Output",
    PROMPT: "Prompt",
    RETRIEVAL: "Retrieval",
    STORAGE: "Storage",
};
export const FlowStatus = {
    FAILED: "Failed",
    NOT_PREPARED: "NotPrepared",
    PREPARED: "Prepared",
    PREPARING: "Preparing",
};
export var FlowValidationDetails;
(function (FlowValidationDetails) {
    FlowValidationDetails.visit = (value, visitor) => {
        if (value.cyclicConnection !== undefined)
            return visitor.cyclicConnection(value.cyclicConnection);
        if (value.duplicateConnections !== undefined)
            return visitor.duplicateConnections(value.duplicateConnections);
        if (value.duplicateConditionExpression !== undefined)
            return visitor.duplicateConditionExpression(value.duplicateConditionExpression);
        if (value.unreachableNode !== undefined)
            return visitor.unreachableNode(value.unreachableNode);
        if (value.unknownConnectionSource !== undefined)
            return visitor.unknownConnectionSource(value.unknownConnectionSource);
        if (value.unknownConnectionSourceOutput !== undefined)
            return visitor.unknownConnectionSourceOutput(value.unknownConnectionSourceOutput);
        if (value.unknownConnectionTarget !== undefined)
            return visitor.unknownConnectionTarget(value.unknownConnectionTarget);
        if (value.unknownConnectionTargetInput !== undefined)
            return visitor.unknownConnectionTargetInput(value.unknownConnectionTargetInput);
        if (value.unknownConnectionCondition !== undefined)
            return visitor.unknownConnectionCondition(value.unknownConnectionCondition);
        if (value.malformedConditionExpression !== undefined)
            return visitor.malformedConditionExpression(value.malformedConditionExpression);
        if (value.malformedNodeInputExpression !== undefined)
            return visitor.malformedNodeInputExpression(value.malformedNodeInputExpression);
        if (value.mismatchedNodeInputType !== undefined)
            return visitor.mismatchedNodeInputType(value.mismatchedNodeInputType);
        if (value.mismatchedNodeOutputType !== undefined)
            return visitor.mismatchedNodeOutputType(value.mismatchedNodeOutputType);
        if (value.incompatibleConnectionDataType !== undefined)
            return visitor.incompatibleConnectionDataType(value.incompatibleConnectionDataType);
        if (value.missingConnectionConfiguration !== undefined)
            return visitor.missingConnectionConfiguration(value.missingConnectionConfiguration);
        if (value.missingDefaultCondition !== undefined)
            return visitor.missingDefaultCondition(value.missingDefaultCondition);
        if (value.missingEndingNodes !== undefined)
            return visitor.missingEndingNodes(value.missingEndingNodes);
        if (value.missingNodeConfiguration !== undefined)
            return visitor.missingNodeConfiguration(value.missingNodeConfiguration);
        if (value.missingNodeInput !== undefined)
            return visitor.missingNodeInput(value.missingNodeInput);
        if (value.missingNodeOutput !== undefined)
            return visitor.missingNodeOutput(value.missingNodeOutput);
        if (value.missingStartingNodes !== undefined)
            return visitor.missingStartingNodes(value.missingStartingNodes);
        if (value.multipleNodeInputConnections !== undefined)
            return visitor.multipleNodeInputConnections(value.multipleNodeInputConnections);
        if (value.unfulfilledNodeInput !== undefined)
            return visitor.unfulfilledNodeInput(value.unfulfilledNodeInput);
        if (value.unsatisfiedConnectionConditions !== undefined)
            return visitor.unsatisfiedConnectionConditions(value.unsatisfiedConnectionConditions);
        if (value.unspecified !== undefined)
            return visitor.unspecified(value.unspecified);
        if (value.unknownNodeInput !== undefined)
            return visitor.unknownNodeInput(value.unknownNodeInput);
        if (value.unknownNodeOutput !== undefined)
            return visitor.unknownNodeOutput(value.unknownNodeOutput);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(FlowValidationDetails || (FlowValidationDetails = {}));
export const FlowValidationSeverity = {
    ERROR: "Error",
    WARNING: "Warning",
};
export const FlowValidationType = {
    CYCLIC_CONNECTION: "CyclicConnection",
    DUPLICATE_CONDITION_EXPRESSION: "DuplicateConditionExpression",
    DUPLICATE_CONNECTIONS: "DuplicateConnections",
    INCOMPATIBLE_CONNECTION_DATA_TYPE: "IncompatibleConnectionDataType",
    MALFORMED_CONDITION_EXPRESSION: "MalformedConditionExpression",
    MALFORMED_NODE_INPUT_EXPRESSION: "MalformedNodeInputExpression",
    MISMATCHED_NODE_INPUT_TYPE: "MismatchedNodeInputType",
    MISMATCHED_NODE_OUTPUT_TYPE: "MismatchedNodeOutputType",
    MISSING_CONNECTION_CONFIGURATION: "MissingConnectionConfiguration",
    MISSING_DEFAULT_CONDITION: "MissingDefaultCondition",
    MISSING_ENDING_NODES: "MissingEndingNodes",
    MISSING_NODE_CONFIGURATION: "MissingNodeConfiguration",
    MISSING_NODE_INPUT: "MissingNodeInput",
    MISSING_NODE_OUTPUT: "MissingNodeOutput",
    MISSING_STARTING_NODES: "MissingStartingNodes",
    MULTIPLE_NODE_INPUT_CONNECTIONS: "MultipleNodeInputConnections",
    UNFULFILLED_NODE_INPUT: "UnfulfilledNodeInput",
    UNKNOWN_CONNECTION_CONDITION: "UnknownConnectionCondition",
    UNKNOWN_CONNECTION_SOURCE: "UnknownConnectionSource",
    UNKNOWN_CONNECTION_SOURCE_OUTPUT: "UnknownConnectionSourceOutput",
    UNKNOWN_CONNECTION_TARGET: "UnknownConnectionTarget",
    UNKNOWN_CONNECTION_TARGET_INPUT: "UnknownConnectionTargetInput",
    UNKNOWN_NODE_INPUT: "UnknownNodeInput",
    UNKNOWN_NODE_OUTPUT: "UnknownNodeOutput",
    UNREACHABLE_NODE: "UnreachableNode",
    UNSATISFIED_CONNECTION_CONDITIONS: "UnsatisfiedConnectionConditions",
    UNSPECIFIED: "Unspecified",
};
export const IngestionJobStatus = {
    COMPLETE: "COMPLETE",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    STARTING: "STARTING",
    STOPPED: "STOPPED",
    STOPPING: "STOPPING",
};
export const IngestionJobFilterAttribute = {
    STATUS: "STATUS",
};
export const IngestionJobFilterOperator = {
    EQ: "EQ",
};
export const IngestionJobSortByAttribute = {
    STARTED_AT: "STARTED_AT",
    STATUS: "STATUS",
};
export const SortOrder = {
    ASCENDING: "ASCENDING",
    DESCENDING: "DESCENDING",
};
export const APISchemaFilterSensitiveLog = (obj) => {
    if (obj.s3 !== undefined)
        return { s3: obj.s3 };
    if (obj.payload !== undefined)
        return { payload: SENSITIVE_STRING };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const CreateAgentActionGroupRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.actionGroupExecutor && { actionGroupExecutor: obj.actionGroupExecutor }),
    ...(obj.apiSchema && { apiSchema: APISchemaFilterSensitiveLog(obj.apiSchema) }),
    ...(obj.functionSchema && { functionSchema: obj.functionSchema }),
});
export const AgentActionGroupFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.actionGroupExecutor && { actionGroupExecutor: obj.actionGroupExecutor }),
    ...(obj.apiSchema && { apiSchema: APISchemaFilterSensitiveLog(obj.apiSchema) }),
    ...(obj.functionSchema && { functionSchema: obj.functionSchema }),
});
export const CreateAgentActionGroupResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agentActionGroup && { agentActionGroup: AgentActionGroupFilterSensitiveLog(obj.agentActionGroup) }),
});
export const GetAgentActionGroupResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agentActionGroup && { agentActionGroup: AgentActionGroupFilterSensitiveLog(obj.agentActionGroup) }),
});
export const UpdateAgentActionGroupRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.actionGroupExecutor && { actionGroupExecutor: obj.actionGroupExecutor }),
    ...(obj.apiSchema && { apiSchema: APISchemaFilterSensitiveLog(obj.apiSchema) }),
    ...(obj.functionSchema && { functionSchema: obj.functionSchema }),
});
export const UpdateAgentActionGroupResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agentActionGroup && { agentActionGroup: AgentActionGroupFilterSensitiveLog(obj.agentActionGroup) }),
});
export const PromptConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.basePromptTemplate && { basePromptTemplate: SENSITIVE_STRING }),
});
export const PromptOverrideConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.promptConfigurations && {
        promptConfigurations: obj.promptConfigurations.map((item) => PromptConfigurationFilterSensitiveLog(item)),
    }),
});
export const AgentFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.instruction && { instruction: SENSITIVE_STRING }),
    ...(obj.customOrchestration && { customOrchestration: obj.customOrchestration }),
    ...(obj.promptOverrideConfiguration && { promptOverrideConfiguration: SENSITIVE_STRING }),
});
export const AgentCollaboratorFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.collaborationInstruction && { collaborationInstruction: SENSITIVE_STRING }),
});
export const AssociateAgentCollaboratorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.collaborationInstruction && { collaborationInstruction: SENSITIVE_STRING }),
});
export const AssociateAgentCollaboratorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agentCollaborator && { agentCollaborator: AgentCollaboratorFilterSensitiveLog(obj.agentCollaborator) }),
});
export const GetAgentCollaboratorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agentCollaborator && { agentCollaborator: AgentCollaboratorFilterSensitiveLog(obj.agentCollaborator) }),
});
export const AgentCollaboratorSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.collaborationInstruction && { collaborationInstruction: SENSITIVE_STRING }),
});
export const ListAgentCollaboratorsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agentCollaboratorSummaries && {
        agentCollaboratorSummaries: obj.agentCollaboratorSummaries.map((item) => AgentCollaboratorSummaryFilterSensitiveLog(item)),
    }),
});
export const UpdateAgentCollaboratorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.collaborationInstruction && { collaborationInstruction: SENSITIVE_STRING }),
});
export const UpdateAgentCollaboratorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agentCollaborator && { agentCollaborator: AgentCollaboratorFilterSensitiveLog(obj.agentCollaborator) }),
});
export const CreateAgentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.instruction && { instruction: SENSITIVE_STRING }),
    ...(obj.customOrchestration && { customOrchestration: obj.customOrchestration }),
    ...(obj.promptOverrideConfiguration && { promptOverrideConfiguration: SENSITIVE_STRING }),
});
export const CreateAgentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agent && { agent: AgentFilterSensitiveLog(obj.agent) }),
});
export const GetAgentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agent && { agent: AgentFilterSensitiveLog(obj.agent) }),
});
export const UpdateAgentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.instruction && { instruction: SENSITIVE_STRING }),
    ...(obj.customOrchestration && { customOrchestration: obj.customOrchestration }),
    ...(obj.promptOverrideConfiguration && { promptOverrideConfiguration: SENSITIVE_STRING }),
});
export const UpdateAgentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.agent && { agent: AgentFilterSensitiveLog(obj.agent) }),
});
export const AgentVersionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.instruction && { instruction: SENSITIVE_STRING }),
    ...(obj.promptOverrideConfiguration && { promptOverrideConfiguration: SENSITIVE_STRING }),
});
export const PatternObjectFilterFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.objectType && { objectType: SENSITIVE_STRING }),
    ...(obj.inclusionFilters && { inclusionFilters: SENSITIVE_STRING }),
    ...(obj.exclusionFilters && { exclusionFilters: SENSITIVE_STRING }),
});
export const PatternObjectFilterConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.filters && { filters: SENSITIVE_STRING }),
});
export const CrawlFilterConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.patternObjectFilter && {
        patternObjectFilter: PatternObjectFilterConfigurationFilterSensitiveLog(obj.patternObjectFilter),
    }),
});
export const ConfluenceCrawlerConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.filterConfiguration && {
        filterConfiguration: CrawlFilterConfigurationFilterSensitiveLog(obj.filterConfiguration),
    }),
});
export const ConfluenceDataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.crawlerConfiguration && {
        crawlerConfiguration: ConfluenceCrawlerConfigurationFilterSensitiveLog(obj.crawlerConfiguration),
    }),
});
export const S3DataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.inclusionPrefixes && { inclusionPrefixes: SENSITIVE_STRING }),
});
export const SalesforceCrawlerConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.filterConfiguration && {
        filterConfiguration: CrawlFilterConfigurationFilterSensitiveLog(obj.filterConfiguration),
    }),
});
export const SalesforceDataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.crawlerConfiguration && {
        crawlerConfiguration: SalesforceCrawlerConfigurationFilterSensitiveLog(obj.crawlerConfiguration),
    }),
});
export const SharePointCrawlerConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.filterConfiguration && {
        filterConfiguration: CrawlFilterConfigurationFilterSensitiveLog(obj.filterConfiguration),
    }),
});
export const SharePointDataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.crawlerConfiguration && {
        crawlerConfiguration: SharePointCrawlerConfigurationFilterSensitiveLog(obj.crawlerConfiguration),
    }),
});
export const WebCrawlerConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.inclusionFilters && { inclusionFilters: SENSITIVE_STRING }),
    ...(obj.exclusionFilters && { exclusionFilters: SENSITIVE_STRING }),
    ...(obj.userAgent && { userAgent: SENSITIVE_STRING }),
    ...(obj.userAgentHeader && { userAgentHeader: SENSITIVE_STRING }),
});
export const WebDataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.crawlerConfiguration && {
        crawlerConfiguration: WebCrawlerConfigurationFilterSensitiveLog(obj.crawlerConfiguration),
    }),
});
export const DataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.s3Configuration && { s3Configuration: S3DataSourceConfigurationFilterSensitiveLog(obj.s3Configuration) }),
    ...(obj.webConfiguration && { webConfiguration: WebDataSourceConfigurationFilterSensitiveLog(obj.webConfiguration) }),
    ...(obj.confluenceConfiguration && {
        confluenceConfiguration: ConfluenceDataSourceConfigurationFilterSensitiveLog(obj.confluenceConfiguration),
    }),
    ...(obj.salesforceConfiguration && {
        salesforceConfiguration: SalesforceDataSourceConfigurationFilterSensitiveLog(obj.salesforceConfiguration),
    }),
    ...(obj.sharePointConfiguration && {
        sharePointConfiguration: SharePointDataSourceConfigurationFilterSensitiveLog(obj.sharePointConfiguration),
    }),
});
export const CreateDataSourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.dataSourceConfiguration && {
        dataSourceConfiguration: DataSourceConfigurationFilterSensitiveLog(obj.dataSourceConfiguration),
    }),
});
export const DataSourceFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.dataSourceConfiguration && {
        dataSourceConfiguration: DataSourceConfigurationFilterSensitiveLog(obj.dataSourceConfiguration),
    }),
});
export const CreateDataSourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.dataSource && { dataSource: DataSourceFilterSensitiveLog(obj.dataSource) }),
});
export const GetDataSourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.dataSource && { dataSource: DataSourceFilterSensitiveLog(obj.dataSource) }),
});
export const UpdateDataSourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.dataSourceConfiguration && {
        dataSourceConfiguration: DataSourceConfigurationFilterSensitiveLog(obj.dataSourceConfiguration),
    }),
});
export const UpdateDataSourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.dataSource && { dataSource: DataSourceFilterSensitiveLog(obj.dataSource) }),
});
export const FlowConditionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.expression && { expression: SENSITIVE_STRING }),
});
export const ConditionFlowNodeConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.conditions && { conditions: obj.conditions.map((item) => FlowConditionFilterSensitiveLog(item)) }),
});
export const ContentBlockFilterSensitiveLog = (obj) => {
    if (obj.text !== undefined)
        return { text: obj.text };
    if (obj.cachePoint !== undefined)
        return { cachePoint: obj.cachePoint };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const MessageFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.content && { content: SENSITIVE_STRING }),
});
export const SystemContentBlockFilterSensitiveLog = (obj) => {
    if (obj.text !== undefined)
        return { text: obj.text };
    if (obj.cachePoint !== undefined)
        return { cachePoint: obj.cachePoint };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const ToolChoiceFilterSensitiveLog = (obj) => {
    if (obj.auto !== undefined)
        return { auto: obj.auto };
    if (obj.any !== undefined)
        return { any: obj.any };
    if (obj.tool !== undefined)
        return { tool: obj.tool };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const ToolConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.tools && { tools: SENSITIVE_STRING }),
    ...(obj.toolChoice && { toolChoice: SENSITIVE_STRING }),
});
export const ChatPromptTemplateConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.messages && { messages: obj.messages.map((item) => MessageFilterSensitiveLog(item)) }),
    ...(obj.system && { system: SENSITIVE_STRING }),
    ...(obj.inputVariables && { inputVariables: SENSITIVE_STRING }),
    ...(obj.toolConfiguration && { toolConfiguration: ToolConfigurationFilterSensitiveLog(obj.toolConfiguration) }),
});
export const TextPromptTemplateConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.text && { text: SENSITIVE_STRING }),
    ...(obj.inputVariables && { inputVariables: SENSITIVE_STRING }),
});
export const PromptTemplateConfigurationFilterSensitiveLog = (obj) => {
    if (obj.text !== undefined)
        return { text: SENSITIVE_STRING };
    if (obj.chat !== undefined)
        return { chat: SENSITIVE_STRING };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const PromptFlowNodeInlineConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.templateConfiguration && {
        templateConfiguration: PromptTemplateConfigurationFilterSensitiveLog(obj.templateConfiguration),
    }),
    ...(obj.inferenceConfiguration && { inferenceConfiguration: obj.inferenceConfiguration }),
});
export const PromptFlowNodeSourceConfigurationFilterSensitiveLog = (obj) => {
    if (obj.resource !== undefined)
        return { resource: obj.resource };
    if (obj.inline !== undefined)
        return { inline: PromptFlowNodeInlineConfigurationFilterSensitiveLog(obj.inline) };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const PromptFlowNodeConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.sourceConfiguration && {
        sourceConfiguration: PromptFlowNodeSourceConfigurationFilterSensitiveLog(obj.sourceConfiguration),
    }),
});
export const FlowNodeConfigurationFilterSensitiveLog = (obj) => {
    if (obj.input !== undefined)
        return { input: obj.input };
    if (obj.output !== undefined)
        return { output: obj.output };
    if (obj.knowledgeBase !== undefined)
        return { knowledgeBase: obj.knowledgeBase };
    if (obj.condition !== undefined)
        return { condition: ConditionFlowNodeConfigurationFilterSensitiveLog(obj.condition) };
    if (obj.lex !== undefined)
        return { lex: obj.lex };
    if (obj.prompt !== undefined)
        return { prompt: PromptFlowNodeConfigurationFilterSensitiveLog(obj.prompt) };
    if (obj.lambdaFunction !== undefined)
        return { lambdaFunction: obj.lambdaFunction };
    if (obj.storage !== undefined)
        return { storage: obj.storage };
    if (obj.agent !== undefined)
        return { agent: obj.agent };
    if (obj.retrieval !== undefined)
        return { retrieval: obj.retrieval };
    if (obj.iterator !== undefined)
        return { iterator: obj.iterator };
    if (obj.collector !== undefined)
        return { collector: obj.collector };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const FlowNodeInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.expression && { expression: SENSITIVE_STRING }),
});
export const FlowNodeFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.configuration && { configuration: FlowNodeConfigurationFilterSensitiveLog(obj.configuration) }),
    ...(obj.inputs && { inputs: obj.inputs.map((item) => FlowNodeInputFilterSensitiveLog(item)) }),
});
export const FlowDefinitionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.nodes && { nodes: obj.nodes.map((item) => FlowNodeFilterSensitiveLog(item)) }),
    ...(obj.connections && { connections: obj.connections.map((item) => item) }),
});
export const CreateFlowRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.definition && { definition: SENSITIVE_STRING }),
});
export const CreateFlowResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.definition && { definition: SENSITIVE_STRING }),
});
export const CreateFlowVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.definition && { definition: SENSITIVE_STRING }),
});
export const GetFlowVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.definition && { definition: SENSITIVE_STRING }),
});
export const DuplicateConditionExpressionFlowValidationDetailsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.expression && { expression: SENSITIVE_STRING }),
});
export const FlowValidationDetailsFilterSensitiveLog = (obj) => {
    if (obj.cyclicConnection !== undefined)
        return { cyclicConnection: obj.cyclicConnection };
    if (obj.duplicateConnections !== undefined)
        return { duplicateConnections: obj.duplicateConnections };
    if (obj.duplicateConditionExpression !== undefined)
        return {
            duplicateConditionExpression: DuplicateConditionExpressionFlowValidationDetailsFilterSensitiveLog(obj.duplicateConditionExpression),
        };
    if (obj.unreachableNode !== undefined)
        return { unreachableNode: obj.unreachableNode };
    if (obj.unknownConnectionSource !== undefined)
        return { unknownConnectionSource: obj.unknownConnectionSource };
    if (obj.unknownConnectionSourceOutput !== undefined)
        return { unknownConnectionSourceOutput: obj.unknownConnectionSourceOutput };
    if (obj.unknownConnectionTarget !== undefined)
        return { unknownConnectionTarget: obj.unknownConnectionTarget };
    if (obj.unknownConnectionTargetInput !== undefined)
        return { unknownConnectionTargetInput: obj.unknownConnectionTargetInput };
    if (obj.unknownConnectionCondition !== undefined)
        return { unknownConnectionCondition: obj.unknownConnectionCondition };
    if (obj.malformedConditionExpression !== undefined)
        return { malformedConditionExpression: obj.malformedConditionExpression };
    if (obj.malformedNodeInputExpression !== undefined)
        return { malformedNodeInputExpression: obj.malformedNodeInputExpression };
    if (obj.mismatchedNodeInputType !== undefined)
        return { mismatchedNodeInputType: obj.mismatchedNodeInputType };
    if (obj.mismatchedNodeOutputType !== undefined)
        return { mismatchedNodeOutputType: obj.mismatchedNodeOutputType };
    if (obj.incompatibleConnectionDataType !== undefined)
        return { incompatibleConnectionDataType: obj.incompatibleConnectionDataType };
    if (obj.missingConnectionConfiguration !== undefined)
        return { missingConnectionConfiguration: obj.missingConnectionConfiguration };
    if (obj.missingDefaultCondition !== undefined)
        return { missingDefaultCondition: obj.missingDefaultCondition };
    if (obj.missingEndingNodes !== undefined)
        return { missingEndingNodes: obj.missingEndingNodes };
    if (obj.missingNodeConfiguration !== undefined)
        return { missingNodeConfiguration: obj.missingNodeConfiguration };
    if (obj.missingNodeInput !== undefined)
        return { missingNodeInput: obj.missingNodeInput };
    if (obj.missingNodeOutput !== undefined)
        return { missingNodeOutput: obj.missingNodeOutput };
    if (obj.missingStartingNodes !== undefined)
        return { missingStartingNodes: obj.missingStartingNodes };
    if (obj.multipleNodeInputConnections !== undefined)
        return { multipleNodeInputConnections: obj.multipleNodeInputConnections };
    if (obj.unfulfilledNodeInput !== undefined)
        return { unfulfilledNodeInput: obj.unfulfilledNodeInput };
    if (obj.unsatisfiedConnectionConditions !== undefined)
        return { unsatisfiedConnectionConditions: obj.unsatisfiedConnectionConditions };
    if (obj.unspecified !== undefined)
        return { unspecified: obj.unspecified };
    if (obj.unknownNodeInput !== undefined)
        return { unknownNodeInput: obj.unknownNodeInput };
    if (obj.unknownNodeOutput !== undefined)
        return { unknownNodeOutput: obj.unknownNodeOutput };
    if (obj.$unknown !== undefined)
        return { [obj.$unknown[0]]: "UNKNOWN" };
};
export const FlowValidationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.details && { details: FlowValidationDetailsFilterSensitiveLog(obj.details) }),
});
export const GetFlowResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.definition && { definition: SENSITIVE_STRING }),
    ...(obj.validations && { validations: obj.validations.map((item) => FlowValidationFilterSensitiveLog(item)) }),
});
export const UpdateFlowRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.definition && { definition: SENSITIVE_STRING }),
});
export const UpdateFlowResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.definition && { definition: SENSITIVE_STRING }),
});
